### PR TITLE
Login Page + Admin Page

### PR DESCRIPTION
**Pull request by andrewdlqnguyen**
Not planning on merging this just yet. Found few errors i need help fixing.

So it works when i build and run it through browser! yay.
login: wedoworkhere
pass: admin

Some changes and error i need help with.

1. Login.html line 19-25
````
<div th:if="${param.error}">
            Invalid username and password.
        </div>
        '<div th:if="${param.logout}">
            You have been logged out.
        </div>
````
I'm getting redline for error and logout. i tried to find a soln online and the only soln i found is to alt+enter and give it a type comment but i dont know what to put for that field.

2. I made config files for the security stuff to see the mapping but I think i can do this all in HomeController and use RequestMapping. I'll have to look into that.

Going to make the admin page interface.

### CODE REVIEW

**Comment by andrewdlqnguyen** [comment]
image

cleaned up login page

**Comment by andrewdlqnguyen** [comment]
Getting weird errors.
image

First image, i resolved some problems by changing error to warning for "No Beans of ... type found"
https://stackoverflow.com/questions/26889970/intellij-incorrectly-saying-no-beans-of-type-found-for-autowired-repository

Second image, i'm not sure how to fix those

-------------------
update
image
  these warnings too. this is after pulling master

**Comment by andrewdlqnguyen** [comment]
image
 maven build fail :c
so many fails... im a failure

**Comment by RyanWork** [comment]
For the "cant wire beans" I have that too for the tests. I wasnt really sure why it happenes though because the tests still run fine eventhough "it cant wire the beans". If it really couldn't then it wouldnt make sense!  I'll look into it because it is my fault. It should not affect your feature though.

**Comment by K-shamG** [comment]
The warnings from data.sql are fine. The tables in the DB are created at runtime since it's using the annotations to create them, this is probably why data.sql doesn't recognize the column names etc until the application is started. You can just ignore those warnings.

**Comment by SupriyaGadigone** [comment]
For the login error I found this: https://docs.spring.io/spring-security/site/docs/current/guides/html5/form-javaconfig.html
It's basically different classes for the for the error to show up I think, not sure if you tried this yet.

**Comment by K-shamG** [comment]
I'm not seeing that error on Login.html for lines 19-25. I tried the gui and that is all working properly too :O Fam what error are you talking about?

**Comment by andrewdlqnguyen** [comment]
> I'm not seeing that error on Login.html for lines 19-25. I tried the gui and that is all working properly too :O Fam what error are you talking about?

For this homedawg
image

when i alt+enter it tells me to put something in the comment for them

**Comment by andrewdlqnguyen** [comment]
> For the login error I found this: https://docs.spring.io/spring-security/site/docs/current/guides/html5/form-javaconfig.html
> It's basically different classes for the for the error to show up I think, not sure if you tried this yet.

I'll have a look at this c:
thanks!

**Comment by andrewdlqnguyen** [comment]
> For the "cant wire beans" I have that too for the tests. I wasnt really sure why it happenes though because the tests still run fine eventhough "it cant wire the beans". If it really couldn't then it wouldnt make sense! I'll look into it because it is my fault. It should not affect your feature though.

I had to change the "error" to display "warning" for those. Ppl are saying its something to do with Jet(intelliJ) not knowing they're perfectly fine. Might be the IDE's problem.

**Comment by K-shamG** [comment]
> > I'm not seeing that error on Login.html for lines 19-25. I tried the gui and that is all working properly too :O Fam what error are you talking about?
> 
> For this homedawg
> image
> 
> when i alt+enter it tells me to put something in the comment for them

Like dis?
https://stackoverflow.com/questions/48511677/how-do-i-handle-collections-with-thymesvar

**Comment by andrewdlqnguyen** [comment]
@adamstaples 
Pull this branch. I don't know why Travis is failing. It is hating on having all the database table to equal NULL but i run locally just fine.

login to Admin Portal (user: wedoworkhere pass: admin)

Currently It is just the interface, there's no mapping.

Going to need to map the Add, Filter, Populate buttons, and the master table to get info from the database based on the Filter we set.

When you click the Add and Filter button, a new modal will pop up to show you what each thing will do.

I think what to do first is to get the master table to populate all the data from database then from there, the filter will break the search down

**Comment by andrewdlqnguyen** [comment]
Things to add:

I couldnt figure out how to add the edit button onto the data rows in master table.. 
Prob helps if it's populated first and then work on having a button appear for each of them. I still can't picture how this will work tho. Any idea?

**Comment by andrewdlqnguyen** [comment]
Reference to #59 
This was the idea for admin control.

-A master table will appear with all topics showing in each column per row
-an add button, a filter button and populate button on right side to give admin action options.

Add Button: simply adds new data

Filter button: shoots out a modal that displays different ways to filter the data to get what the admin specifically want. Presses okay and brings back to the master table view. (Maybe a counter on the side that'll display to give awareness how many data will appear if they press populate)

Populate button: Displays all the data (populates the master table) depending on the filter and an edit button will appear on the side of those data rows.

I think this way if an admin goes in blind, they can find what they want to edit in general.

File: src/main/java/SYSC4806/Configurations/WebSecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                .antMatchers("/", "/home").permitAll()
+                .antMatchers( "/css/**", "/js/**", "/img/**", "/webjars/**").permitAll()
```
Comment (adamstaples) [inline src/main/java/SYSC4806/Configurations/WebSecurityConfig.java]: This does not include the results url, which is what we use to display our table when we submit the search results. Adding a "/results/**" will fix it I believe.

**Review by adamstaples — CHANGES_REQUESTED** [review]

**Review by adamstaples — APPROVED** [review]

**Pull request merged**